File: Motocitizen/src/motocitizen/fragments/ConfirmDialog.java
package motocitizen.fragments;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.os.Bundle;

public class ConfirmDialog extends DialogFragment {

    public static ConfirmDialog newInstance(String title, String positiveButton, String negativeButton) {
        ConfirmDialog dialogFragment = new ConfirmDialog();
        Bundle        bundle         = new Bundle();
        bundle.putString("title", title);
        bundle.putString("positive", positiveButton);
        bundle.putString("negative", negativeButton);
        dialogFragment.setArguments(bundle);
        return dialogFragment;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder res = new AlertDialog.Builder(getActivity());
        res.setTitle(getArguments().getString("title"));
        res.setIcon(android.R.drawable.ic_dialog_alert);
        String positive = getArguments().getString("positive");
        String negative = getArguments().getString("negative");
        if (positive != null && !positive.isEmpty())
            res.setPositiveButton(positive, (dialog, whichButton) -> getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, getActivity().getIntent()));
        if (negative != null && !negative.isEmpty())
            res.setNegativeButton(negative, (dialog, whichButton) -> getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_CANCELED, getActivity().getIntent()));
        return res.create();
    }
}

File: Motocitizen/src/motocitizen/fragments/ConfirmDialog.kt
package motocitizen.fragments

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.content.res.Resources
import android.os.Bundle

class ConfirmDialog() : DialogFragment() {
    private val YES_STRING = android.R.string.yes
    private val NO_STRING = android.R.string.no
    private val ICON = android.R.drawable.ic_dialog_alert

    private lateinit var title: String

    constructor(title: String) : this() {
        this.title = title
    }

    override fun onCreateDialog(savedInstanceState: Bundle): Dialog {
        val dialog = AlertDialog.Builder(activity)
        dialog.setTitle(title)
        dialog.setIcon(ICON)
        dialog.setPositiveButton(Resources.getSystem().getString(YES_STRING)) { _, _ -> targetFragment.onActivityResult(targetRequestCode, Activity.RESULT_OK, activity.intent) }
        dialog.setNegativeButton(Resources.getSystem().getString(NO_STRING)) { _, _ -> targetFragment.onActivityResult(targetRequestCode, Activity.RESULT_CANCELED, activity.intent) }
        return dialog.create()
    }
}
